File: Try_GameEngine5/src/com/example/try_gameengine/framework/BaseActivity.java
package com.example.try_gameengine.framework;

import java.util.Iterator;

import android.os.Bundle;
import android.app.Activity;
import android.util.DisplayMetrics;
import android.view.Menu;

public abstract class BaseActivity extends Activity {
	private IGameModel gameModel;
	private IGameController gameController;
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
//		setContentView(R.layout.activity_main);
		
		DisplayMetrics dm = new DisplayMetrics();
		getWindowManager().getDefaultDisplay().getMetrics(dm);
		
		CommonUtil.screenHeight = dm.heightPixels;
		CommonUtil.screenWidth = dm.widthPixels;
		
		BitmapUtil.initBitmap(this);
		
//		LayerManager.initLayerManager();
		LayerManager.setNoSceneLayer();
		
		initGameModel();
		initGameController();
	}
	
	protected abstract void initGameModel();
	
	protected abstract void initGameController();

}

File: Try_GameEngine5/src/com/example/try_gameengine/framework/BaseActivity.kt
package com.example.try_gameengine.framework

import android.app.Activity
import android.os.Bundle
import android.util.DisplayMetrics

abstract class BaseActivity : Activity() {
    private val gameModel: IGameModel? = null
    private val gameController: IGameController? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        //		setContentView(R.layout.activity_main);
        val dm = DisplayMetrics()
        getWindowManager().getDefaultDisplay().getMetrics(dm)

        CommonUtil.screenHeight = dm.heightPixels
        CommonUtil.screenWidth = dm.widthPixels

        BitmapUtil.initBitmap(this)


//		LayerManager.getInstance().initLayerManager.getInstance()();
//		LayerManager.getInstance().setNoSceneLayer();
        initGameModel()
        initGameController()
    }

    protected abstract fun initGameModel()

    protected abstract fun initGameController()
}

